oxidation/client/tests/unit/HomePage.spec.ts

```diff
@@ -128,25 +130,111 @@ describe('HomePage.vue', () => {
       expect(wrapper.vm.selectedDevice).toEqual({
-        organizationId: 'MegaShark',
-        humanHandle: 'Maxime Grandcolas',
+        organizationId: 'Black Mesa',
+        humanHandle: 'Dr. Gordon Freeman',
         deviceLabel: 'device_label',
         keyFilePath: 'key_file_path',
         deviceId: 'device_id',
-        slug: 'slug1',
+        slug: 'slug3',
         ty: {tag: 'Password'}
       });
     });
   });
 
+  describe('Login filter orgs', () => {
+    const wrapper = mount(HomePage, {
+      global: {
+        plugins: [i18n]
+      }
+    });
+
+    it('should filter orgs', async () => {
+      expect(wrapper.vm.deviceList.length).toEqual(7);
+      expect(wrapper.vm.orgSearchString).toEqual('');
+      expect(wrapper.vm.sortBy).toEqual('organization');
+      expect(wrapper.vm.sortByAsc).toEqual(true);
+      expect(wrapper.vm.filteredDevices.length).toEqual(wrapper.vm.deviceList.length);
+
+      // Should be ordered by ascending org name by default
+      expect(wrapper.vm.filteredDevices[0].organizationId).toEqual('Black Mesa');
+      expect(wrapper.vm.filteredDevices[6].organizationId).toEqual('Sanctum Sanctorum');
+
+      expect(wrapper.findComponent('ion-input').exists()).toBeTruthy();
+      await wrapper.findComponent('ion-input').setValue('la');
+
+      expect(wrapper.vm.orgSearchString).toEqual('la');
+      expect(wrapper.vm.filteredDevices.length).toEqual(3);
+      expect(wrapper.vm.filteredDevices).toEqual([{
+        organizationId: 'Black Mesa',
+        humanHandle: 'Dr. Gordon Freeman',
+        deviceLabel: 'device_label',
+        keyFilePath: 'key_file_path',
+        deviceId: 'device_id',
+        slug: 'slug3',
+        ty: {tag: 'Password'}
+      }, {
+        organizationId: 'Planet Express Is The Best Comp!',
+        humanHandle: 'Dr. John A. Zoidberg',
+        deviceLabel: 'device_label',
+        keyFilePath: 'key_file_path',
+        deviceId: 'device_id',
+        slug: 'slug1',
+        ty: {tag: 'Password'}
+      },
+      {
+        organizationId: 'Riviera M.D.',
+        humanHandle: 'Dr. Nicholas "Nick" Riviera',
+        deviceLabel: 'device_label',
+        keyFilePath: 'key_file_path',
+        deviceId: 'device_id',
+        slug: 'slug7',
+        ty: {tag: 'Password'}
+      }]);
+
+      // Resetting the search string
+      await wrapper.findComponent('ion-input').setValue('');
+      expect(wrapper.vm.orgSearchString).toEqual('');
+      expect(wrapper.vm.filteredDevices.length).toEqual(7);
+
+      // Inverting the sort order
+      wrapper.vm.sortByAsc = false;
+
+      // Should be ordered by descending org name
+      expect(wrapper.vm.filteredDevices[0].organizationId).toEqual('Sanctum Sanctorum');
+      expect(wrapper.vm.filteredDevices[6].organizationId).toEqual('Black Mesa');
+
+      wrapper.vm.sortBy = 'last_login';
+      // Should be order by last login date descending
+      expect(wrapper.vm.filteredDevices[0].organizationId).toEqual('OsCorp');
```

Comment:
use slug instead of organizationId, it's more readable as lastLogin date is attach to slugs.
+ missing test on descending, first org as a value whereas org 6 has not, at lest test 4 firsts values.

Reply:
done